woocommerce/woocommerce-android
# Delete custom amounts business logic

Closes: #10053 


#### NOTE: Please follow the below PR orders for reviewing
This one!
https://github.com/woocommerce/woocommerce-android/pull/10057
https://github.com/woocommerce/woocommerce-android/pull/10062

### Description

This PR introduces functionality that allows users to remove previously added custom amounts. With this enhancement, managing custom amounts becomes more flexible and user-friendly.

**Key Changes:**

**Removal Functionality:** Added the core logic to facilitate the deletion of existing custom amounts, ensuring that removed amounts are no longer displayed or used within the application.

**UI Update:** Integrated a user-friendly interface that provides clear visual cues and actions for removing custom amounts.

### Testing instructions

1. Navigate to the `orders` tab
2. Click on "+" button to navigate the the order creation screen
3. Click on the "Add custom amount" button
4. Add a custom amount
5. Custom amount gets added
7. Click on the cross icon on the custom amount row
8. Ensure the selected custom amount row is removed and appropriate changes are reflected in payments section

### Images/gif


https://github.com/woocommerce/woocommerce-android/assets/1331230/6f2c19f2-cf97-40f8-8430-8b3346f6f011
